.github/workflows/benchstat.yml

```diff
@@ -0,0 +1,100 @@
+
+name: benchstat
+
+on:
+  - pull_request
+
+permissions:
+  contents: read
+  issues: write
+  pull-requests: write
+
+jobs:
+  benchstat:
+    strategy:
+      matrix:
+        go: ["1.21"]
+        platform: [ubuntu-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+        with:
+          go-version: ${{ matrix.go }}
+          cache: false
+      - name: Checkout out code
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+        with:
+          fetch-depth: '0'
+      - name: Determine go cache key
+        id: go-cache-key
+        run: |
+          echo "key=${{ runner.os }}-go-${{ matrix.go }}-${{ hashFiles('**/go.sum', './Makefile') }}" >> "$GITHUB_OUTPUT"
+          echo "restore-key=${{ runner.os }}-go-${{ matrix.go }}" >> "$GITHUB_OUTPUT"
+      - name: Determine Go cache paths
+        id: go-cache-paths
+        run: |
+          echo "go-build=$(go env GOCACHE)" >> "$GITHUB_OUTPUT"
+          echo "go-mod=$(go env GOMODCACHE)" >> "$GITHUB_OUTPUT"
+          echo "go-bin=$(go env GOPATH)/bin" >> "$GITHUB_OUTPUT"
+      - name: Cache
+        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+        with:
+          path: |
+            ${{ steps.go-cache-paths.outputs.go-mod }}
+            ${{ steps.go-cache-paths.outputs.go-bin }}
+            ${{ steps.go-cache-paths.outputs.go-build }}
+          key: ${{ steps.go-cache-key.outputs.key }}
+          restore-keys: |
+            ${{ steps.go-cache-key.outputs.restore-key }}
+      - name: Install tools
+        run: |
+          go mod download
+          make tools
+      - name: Bench New
+        run: |
+          go test -bench=. -count=10 -run=^# | tee new.txt
```

Comment:
What does `-run=&^#` do? Can you add a comment?

Reply:
The `-run` option allows for specifying a regex to match against tests to run specific tests. In this cast `^#` should not match *any* tests, thus making so only the benchmarks run, not any other go tests.

I added a comment to hopefully clarify this.

Reply:
Do we not want `-run=^$` if we want a regex that matches nothing?

Reply:
That would also work.